pdf_converter.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sun Dec  8 00:48:01 2019
+
+@author: Quang
+"""
+
+import codecs
+from run import _get_title_from_file_path, _chapter_path_from_chapter_number
+import os
+import shutil
+import pdfkit
+
+# On linux we don't have to initialize config. This config path apllies only on Window
+config = pdfkit.configuration(wkhtmltopdf="C:\\Program Files\\wkhtmltopdf\\bin\\wkhtmltopdf.exe")
+
+PARTS = [
+    {'path': './chapters/p00_01_04.md', 'range': [1, 4]},
+    {'path': './chapters/p01_05_12.md', 'range': [5, 12]},
+    {'path': './chapters/p02_13_19.md', 'range': [13, 19]},
+    {'path': './chapters/p03_20_27.md', 'range': [20, 27]},
+    {'path': './chapters/p04_28_32.md', 'range': [28, 32]},
+    {'path': './chapters/p05_33_35.md', 'range': [33, 35]},
+    {'path': './chapters/p06_36_43.md', 'range': [36, 43]},
+    {'path': './chapters/p07_44_46.md', 'range': [44, 46]},
+    {'path': './chapters/p08_47_52.md', 'range': [47, 52]},
+    {'path': './chapters/p09_53_57.md', 'range': [53, 57]},
+    {'path': './chapters/p10_58.md', 'range': [58, 58]},
+]
+
+no_part_list =['p{:02d}'.format(i) for i in range(0, 11)]
+
+no_chapter_list = ['{:02d}'.format(i) for i in range(1, 59)]
+        
+# extract list of all part titles and chapter titles
+part_list = []
+chapter_list = []
+
+def _convert_title_to_link(title):
+    title = title.lower()
+    title = title.replace(" ","-")
+    title = title.replace(".","")
+    title = title.replace(":","")
+    title = title.replace("/","")
+    title = title.replace("?","")
+    title = title.replace(",","")
+    title = title.replace("#-","#user-content-")
+    return title
+
+for part in PARTS:
+    part_path = part['path']
+    # Extract the original parth title
+    part_title = _get_title_from_file_path(part_path)
+    
+    # Convert to the html link syntax
+    part_list.append(_convert_title_to_link(part_title))
+    
+    start_chapter, end_chatper = part['range']
+    for chapter_number in range(start_chapter, end_chatper + 1):
+        chapter_path = _chapter_path_from_chapter_number(chapter_number)
+        
+        # Extract the original chapter title
+        chapter_title = _get_title_from_file_path(chapter_path)
+        # Convert to html link syntax
+        chapter_list.append(_convert_title_to_link(chapter_title))
+  
+
+All_chapters = "./chapters/all_chapters"
```

Comment:
Viết hoa toàn bộ chứ em :)

Reply:
He he, sorry anh em học robotic nên viết code chưa được clean lắm :smile: